Expand helm requirements page to explain matching the helm policy (#1943)
rancher/rancher-docs: docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md

<DeprecationHelm2 />

> ## Identifying the proper Helm 3 version

Reviewer: ```suggestion
## Identifying the Proper Helm v3 Version
```

> Select any Helm 3 version that is officially compatible with the Kubernetes version range you are using from our supported list.

Reviewer: ```suggestion
Select any Helm v3 version that is officially compatible with the Kubernetes version range you are using from our [Rancher Support Matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions).
```

> To apply this rule, you may need to reference two external resources:
> - **Helm Version Compatibility:** Refer to the [Helm Version Support Policy](https://helm.sh/docs/topics/version_skew/) and select the version matching the rule for your Rancher minor target.

Reviewer: ```suggestion
To apply this rule, you may need to reference two external resources:

- **Helm Version Compatibility:** Refer to the [Helm Version Support Policy](https://helm.sh/docs/topics/version_skew/) and select the version matching the rule for your Rancher minor target.
```

> - **Rancher's Kubernetes Support Range:** Use the [Rancher Support Matrix](https://rancher.com/support-maintenance-terms/) to identify the Kubernetes versions supported by your target Rancher minor version.

Reviewer: ```suggestion
- **Rancher's Kubernetes Support Range:** Use the [Rancher Support Matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions) to identify the Kubernetes versions supported by your target Rancher minor version.
```

> ### Example:
> - **Scenario:** You are targeting Rancher 2.11.4, which supports Kubernetes versions 1.30 through 1.32

Reviewer: ```suggestion
### Example

- **Scenario:** You are targeting Rancher v2.11.4, which supports Kubernetes versions 1.30 through 1.32.
```

> - **Application:** Our rule requires a Helm version that supports this range. You can verify this by checking the Helm version's compatibility with the highest version in the range, Kubernetes 1.32.

Reviewer: ```suggestion
- **Application:** Our rule requires a Helm version that supports this range. You can verify this by checking the Helm version's compatibility with the highest version in the range, Kubernetes v1.32.
```

> - **Result:** You find that both Helm 3.17 and Helm 3.18 support the Kubernetes 1.30-1.32 range.

Reviewer: ```suggestion
- **Result:** You find that both Helm v3.17 and Helm v3.18 support the Kubernetes v1.30-v1.32 range.
```

> - Both will work, however selecting 3.18 is preferred because it is the newest Helm minor with overlapping support range.

Reviewer: ```suggestion
- Although both work, we recommend Helm v3.18 because it is the newest Helm minor version overlapping the supported Kubernetes range.
```

## Additional Notes

- Helm v3.2.x or higher is required to install or upgrade Rancher v2.5.

> - Helm v2 support was removed in Rancher v2.9.z

Reviewer: ```suggestion
- Helm v2 support was removed in Rancher v2.9.x.
```
